File: Hyperdrive.Service/Hyperdrive.Tier.ViewModels/Classes/Views/ViewApplicationUserDriveItem.cs
using System;
using System.Xml.Serialization;

using Hyperdrive.Tier.ViewModels.Interfaces.Views;

namespace Hyperdrive.Tier.ViewModels.Classes.Views
{
    [XmlRoot("application-user-archive")]
    public class ViewApplicationUserDriveItem : IViewBase, IViewKey
    {
        public ViewApplicationUserDriveItem()
        {
        }

        [XmlElement("id")]
        public int Id { get; set; }

        [XmlElement("last-modified")]
        public DateTime LastModified { get; set; }

        [XmlElement("archive")]
        public virtual ViewDriveItem DriveItem { get; set; }

        [XmlElement("application-user")]
        public virtual ViewApplicationUser ApplicationUser { get; set; }
    }
}

File: Hyperdrive.Client/src/viewmodels/views/viewapplicationuserdriveitem.ts
import { ViewBase } from './viewbase';
import { ViewKey } from './viewkey';
import { ViewDriveItem } from './viewdriveitem';
import { ViewApplicationUser } from './viewapplicationuser';

export interface ViewApplicationUserDriveItem extends ViewKey, ViewBase {
    DriveItem: ViewDriveItem;
    ApplicationUser: ViewApplicationUser;
}
